### PR TITLE
[NBS] Updated Search Docs

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

This updates the vast majority of the Search docs to work with the new backend system in a more visible way.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by awanlin** [comment]
Not stale just waiting for a review but I think summer vacations are slowing things down 👍

**Comment by benjdlambert** [comment]
@backstage/search-maintainers can you take a peek at this also please :pray:

File: mkdocs.yml
Comment (emmaindal) [inline mkdocs.yml:73]: Hey @awanlin awesome to see some docs updates. More then happy to review these and update the docs structure, just wanted to make sure you have seen the concept docs we have which is meant to be there to explain different concepts where we have a section for collators https://backstage.io/docs/features/search/concepts#collators. I wonder if we now have some overlap between the concept docs and how to guides with these changes you have made. It also looks like the new collators docs include how to guides so I'm interested in knowing why you wanted to separate them out. Thank you!

Comment (awanlin) [reply mkdocs.yml:73]: It really came down to needing a place to make this content more visible as it was coming up on Discord regularly and following what was done with Search Engines. The Concepts page seemed like the wrong place to add installation and configuration information and the How To Guides is very long and not as easy to find what you are looking for as it could be.

File: docs/features/search/collators.md
Comment (awanlin) [inline docs/features/search/collators.md:7]: @emmaindal, I've removed the line from this opening section that was already in the Concepts page and linked to that section. Does that seem better to you?

**Review by emmaindal — APPROVED** [review]
Yes I'm happy to get these docs updates in, we don't spend much time on improving the docs atm so really appreciate you opening this PR! 🙏

File: mkdocs.yml
Comment (emmaindal) [reply mkdocs.yml:73]: Makes sense to me, lets try it out! Thanks a lot!

**Comment by awanlin** [comment]
Hi @Rugvip, looks like I just need approval from you now, I know you have a long backlog though. Just want to highlight that these are docs changes only.

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**
